src/auth/baseexternalclient.ts

```diff
@@ -20,6 +20,8 @@ import {
   GaxiosResponse,
 } from 'gaxios';
 import * as stream from 'stream';
+import {log as makeLog} from 'google-logging-utils';
+const log = makeLog('auth');
```

Comment:
I think it may be cleaner and flexible to add this to the base `AuthClient` class and have it as a member or static method.

This will allow:
- Customers to change/modify the log functionality (if they want)
- Dedupe the import statements across the files

Additionally, in the next major we're planning to unify the `Transporter` instance into `AuthClient` via this PR:
- https://github.com/googleapis/google-auth-library-nodejs/pull/1860/files#diff-1be86cf1a764f43a299d2d037e8db93a74904846592cfe5f31c0981079c08992R256

It would be an even smaller migration after that has landed as we could extend the `GaxiosOptions` interface with a `Symbol` (which would note the type of log) and use `log.info` on every request (as all requests would use that `_request` method).

Reply:
Thanks for the quick review. I'll make that change tomorrow.

Reply:
A lot of the logging can be done through AuthClient now, good call. There are a few objects that don't seem to derive ultimately from `AuthClient`, so I moved makeLog() into the class itself to make it cleaner. (`makeLog` caches the loggers by name, so there's little harm in calling it multiple times.)